frontend/apps/marketing/src/components/link/__tests__/Link.test.tsx

```diff
@@ -0,0 +1,53 @@
+import Link, {LinkProps} from '@/components/link';
+import {render, screen} from '@testing-library/react';
+
+describe('Link Component', () => {
+  const defaultProps = {
+    href: 'https://example.com',
+    color: 'purple',
+    size: 'm',
+    bottomMargin: 'none',
+    isLinkExternal: false,
+    children: 'Test Link',
+  } as LinkProps;
+
+  it('renders the Link component with correct text', () => {
+    render(<Link {...defaultProps} />);
+    expect(screen.getByText('Test Link')).toBeInTheDocument();
+  });
+
+  it('renders with correct href attribute', () => {
+    render(<Link {...defaultProps} />);
+    expect(screen.getByRole('link')).toHaveAttribute('href', defaultProps.href);
+  });
+
+  it('applies correct CSS classes based on props', () => {
+    render(<Link {...defaultProps} />);
+    const linkElement = screen.getByRole('link');
+    expect(linkElement).toHaveClass('link');
+    expect(linkElement).toHaveClass('link-purple');
+    expect(linkElement).toHaveClass('link-m');
+    expect(linkElement).toHaveClass('link-bottomMargin-none');
+  });
```

Comment:
Do we want to test for CSS classes? I remember you saying we try not to do this [here](https://github.com/code-dot-org/code-dot-org/pull/63191#discussion_r1915449668). Or is this different since it's testing the props?

Reply:
Good question!
Reason for different approach between these two is caused by these two being two separate instances with different tools.

For `component-library` we have storybook and eyes tests, for `marketing` we currently only have jest tests. I feel like we're planning to also add eyes tests to marketing, but not sure if it'll be covering link component specifically. (It probably will, but when?)

That being said, I'm happy to remove this test, it's just that I don't see any other ways to actually automatically test if our marketing custom functionality is working correctly. We can discuss it on checkin